src/Microsoft.Health.Dicom.Api/Features/Filters/DicomRequestContextRouteDataPopulatingFilterAttribute.cs

```diff
@@ -39,7 +39,20 @@ public override void OnActionExecuting(ActionExecutingContext context)
         IDicomRequestContext dicomRequestContext = _dicomRequestContextAccessor.RequestContext;
         dicomRequestContext.RouteName = context.ActionDescriptor?.AttributeRouteInfo?.Name;
 
-        dicomRequestContext.Version = context.HttpContext.GetRequestedApiVersion()?.MajorVersion;
+        ApiVersion version = null;
+        try
+        {
+            version = context.HttpContext?.GetRequestedApiVersion();
+        }
+        catch (ArgumentNullException)
```

Comment:
Yeesh, I definitely don't love this lol

Reply:
Spoke offline - updated fix to look more like https://github.com/microsoft/dicom-server/commit/d36e85fea4e22f2b6e783f14f29a07ca43aba45f